### PR TITLE
Isiimportertest

### PR DESCRIPTION
**Pull request by obraliar**

### CODE REVIEW

**Comment by lenhard** [comment]
Please rebase this on master. This should also get the build to work on CircleCI.

File: src/test/java/net/sf/jabref/importer/fileformat/IsiImporterTest.java [hunk 1]
```diff
@@ -88,152 +134,175 @@ public void testProcessSubSup() {
...
 
-        // What todo with PD and UT?
+            Assert.assertEquals(BibtexEntryTypes.ARTICLE, entry.getType());
+            Assert.assertEquals("Optical Materials", entry.getField("journal"));
+            Assert.assertEquals("2006", entry.getField("year"));
+            Assert.assertEquals("28", entry.getField("volume"));
+            Assert.assertEquals("5", entry.getField("number"));
+            Assert.assertEquals("467--72", entry.getField("pages"));
+
+            // What todo with PD and UT?
```
Comment (simonharrer) [inline src/test/java/net/sf/jabref/importer/fileformat/IsiImporterTest.java]: What about this comment?

File: src/test/java/net/sf/jabref/importer/fileformat/IsiImporterTest.java [hunk 2]
```diff
@@ -88,152 +134,175 @@ public void testProcessSubSup() {
...
+            Assert.assertEquals("#jul#", a.getField("month"));
+            Assert.assertEquals("89", a.getField("volume"));
+            Assert.assertEquals("4", a.getField("number"));
+
+            // JI Appl. Phys. Lett. (USA)
+
+            // BP 41103-1
+            // EP 41103-41103-3
+            // PS 41103-1-3
+            // assertEquals("41103-1-3", a.getField("pages"));
```
Comment (simonharrer) [inline src/test/java/net/sf/jabref/importer/fileformat/IsiImporterTest.java]: Why is this commented out?

File: src/test/java/net/sf/jabref/importer/fileformat/IsiImporterTest.java [hunk 3]
```diff
@@ -88,152 +134,175 @@ public void testProcessSubSup() {
...
+             * frequency conversion, parametric oscillation and amplification A4255R
+             * Lasing action in other solids A4260B Design of specific laser systems
+             * B4340K Optical harmonic generation, frequency conversion, parametric
+             * oscillation and amplification B4320G Solid lasers/ wavelength
+             * 2.78E-07 m; wavelength 5.32E-07 m; wavelength 2.66E-07 m; power
+             * 2.4E-05 W; size 8.0E-03 m; power 1.53E-01 W; wavelength 5.56E-07 m/
+             * BaB2O4/ss B2/ss Ba/ss O4/ss B/ss O/ss C1 Degl'Innocenti, R.; Guarino,
+             * A.; Poberaj, G.; Gunter, P.; Nonlinear Opt. Lab., Inst. of Quantum
+             * Electron., Zurich, Switzerland
+             */
```
Comment (simonharrer) [inline src/test/java/net/sf/jabref/importer/fileformat/IsiImporterTest.java]: Why is this long comment in the code? There are also other comments there. Why are they necessary? Or can they be removed?

File: src/test/java/net/sf/jabref/importer/fileformat/IsiImporterTest.java [hunk 4]
```diff
@@ -273,97 +342,167 @@ public void testGetIsCustomImporter() {
...
+            Assert.assertEquals("1545-598X", a.getField("SN"));
 
+            Assert.assertEquals("387--391", a.getField("pages"));
+
+            Assert.assertEquals("Gamba, P. and Dell'Acqua, F. and Lisini, G.", a.getField("author"));
+
+            Assert.assertEquals("2006", a.getField("year"));
+
+            Assert.assertEquals("Perceptual grouping, street extraction, urban remote sensing", a.getField("keywords"));
+
```
Comment (simonharrer) [inline src/test/java/net/sf/jabref/importer/fileformat/IsiImporterTest.java]: Try to use empty lines to group statements together. Like in a book where a paragraph groups together sentences.

**Comment by zellerdev** [comment]
I didn't created this Test. These comments where already there and I didn't touched them while I tested the rest of it.

**Comment by simonharrer** [comment]
Even if you did not create these tests, please have a look and either explain the comments or delete them. 

Another issue: the tests do not cover 100% of the importer. Have a look at https://codecov.io/github/JabRef/jabref/src/main/java/net/sf/jabref/importer/fileformat/IsiImporter.java?ref=2d237028d5494f4533344acf5b232c08d3170ae9

**Comment by koppor** [comment]
Coverage increased to [98,01%](https://codecov.io/github/JabRef/jabref/src/main/java/net/sf/jabref/importer/fileformat/IsiImporter.java?ref=27a0b94ecf2df2c005adf409b6e03847db39ffbd). [Line 280](https://codecov.io/github/JabRef/jabref/src/main/java/net/sf/jabref/importer/fileformat/IsiImporter.java?ref=27a0b94ecf2df2c005adf409b6e03847db39ffbd#l-280) seems to be difficult to cover, the others should be manageable.

**Comment by koppor** [comment]
Current coverage [98,01%](https://codecov.io/github/JabRef/jabref/src/main/java/net/sf/jabref/importer/fileformat/IsiImporter.java?ref=27a0b94ecf2df2c005adf409b6e03847db39ffbd). We merge nevertheless. If possible, create a **new pull request** increasing the coverage again.

**Pull request merged**